server/services/lduService.js

```diff
@@ -28,16 +29,20 @@ module.exports = function createLduService(deliusClient, activeLduClient) {
 
       const { content: ldus = [] } = await deliusClient.getAllLdusForProbationArea(probationAreaCode)
 
-      const allLdus = ldus.map(ldu => ({
-        code: ldu.code,
-        description: ldu.description,
-        active: activeLdusCodes.includes(ldu.code),
-      }))
+      const allLdus = ldus
+        .map(ldu => ({
+          code: ldu.code,
+          description: ldu.description,
+          active: activeLdusCodes.includes(ldu.code),
+        }))
+        .sort((a, b) => a.description.localeCompare(b.description, 'en', { ignorePunctuation: true }))
+
+      const uniqueLdus = Object.entries(R.groupBy(ldu => ldu.code, allLdus)).map(ldu => ldu[1][0])
```

Comment:
There's no ordering on Object.entries so your previous sort is lost.
Also the ldu's are in an arbitrary order within each group.  They need to be sorted by description before selecting the first.

Try something like this:
```javascript
const lduMap = R.groupBy(R.prop('code'), ldus)
Object
  .entries(lduMap)
  .sort((a, b) => a[0].localeCompare(b[0]))
  .map(entry => entry[1].sort((a,b) => a.description.localeCompare(b.description))[0])
```
Here's an [example of how that works](https://ramdajs.com/repl/#?const%20ldus%20%3D%20%5B%0A%20%20%7B%20code%3A%20%27A%27%2C%20description%3A%20%27An%20A%27%7D%2C%0A%20%20%7B%20code%3A%20%27C%27%2C%20description%3A%20%27A%20c%27%20%7D%2C%0A%20%20%7B%20code%3A%20%27B%27%2C%20description%3A%20%27A%20B%27%20%7D%2C%0A%20%20%7B%20code%3A%20%27A%27%2C%20description%3A%20%27A%20a%27%20%7D%2C%0A%20%20%7B%20code%3A%20%27A%27%2C%20description%3A%20%27An%20a%27%7D%2C%0A%20%20%7B%20code%3A%20%27B%27%2C%20description%3A%20%27A%20b%27%20%7D%2C%0A%20%20%7B%20code%3A%20%27A%27%2C%20description%3A%20%27A%20A%27%20%7D%2C%0A%5D%0A%0Aconst%20lduMap%20%3D%20R.groupBy%28R.prop%28%27code%27%29%2C%20ldus%29%0AObject%0A%20%20.entries%28lduMap%29%0A%20%20.sort%28%28a%2C%20b%29%20%3D%3E%20a%5B0%5D.localeCompare%28b%5B0%5D%29%29%0A%20%20.map%28entry%20%3D%3E%20entry%5B1%5D.sort%28%28a%2Cb%29%20%3D%3E%20a.description.localeCompare%28b.description%29%29%5B0%5D%29%0A) That you can experiment with.

I'd suggest extracting the code which transforms `allLdus' into `uniqueLdus` into a function, then writing some tests for that function to show that it does what you expect of it.

Reply:
done